source/plugins/go/input/lib/kubernetescontainerinventory.go

```diff
@@ -0,0 +1,519 @@
+			} else {
+				if containerID == "" || containerRuntime == "" {
+					containerInventoryRecord["EnvironmentVar"] = ""
+				} else {
+					if strings.EqualFold(containerRuntime, "cri-o") {
+						containerInventoryRecord["EnvironmentVar"] = obtainContainerEnvironmentVars(fmt.Sprintf("crio-%s", containerID))
+					} else {
+						containerInventoryRecord["EnvironmentVar"] = obtainContainerEnvironmentVars(containerID)
+					}
+				}
+			}
+
+			containerInventoryRecords = append(containerInventoryRecords, containerInventoryRecord)
+		}
+	}
+
+	return containerInventoryRecords
+}
+
+func getContainersInfoMap(podItem map[string]interface{}, isWindows bool) map[string]map[string]string {
+	containersInfoMap := make(map[string]map[string]string)
+
+	nodeName := ""
+	if val, ok := podItem["spec"].(map[string]interface{})["nodeName"].(string); ok {
+		nodeName = val
+	}
+
+	createdTime := podItem["metadata"].(map[string]interface{})["creationTimestamp"].(string)
+	podName := podItem["metadata"].(map[string]interface{})["name"].(string)
+	namespace := podItem["metadata"].(map[string]interface{})["namespace"].(string)
+
+	if len(podItem) > 0 && podItem["spec"] != nil && len(podItem["spec"].(map[string]interface{})) > 0 {
+		containersField, found := podItem["spec"].(map[string]interface{})["containers"]
+		if !found {
+			FLBLogger.Printf("KubernetesContainerInventory::getContainersInfoMap: containers field not found in podItem")
+			return containersInfoMap
+		}
+		podContainers, ok := containersField.([]interface{})
+		if !ok {
+			return containersInfoMap
+		}
+		initContainersField, found := podItem["spec"].(map[string]interface{})["initContainers"]
+		initContainers, ok := initContainersField.([]interface{})
+		if found && ok {
+			podContainers = append(podContainers, initContainers...)
+		}
+
+		if len(podContainers) > 0 {
+			for _, container := range podContainers {
+				containerMap, ok := container.(map[string]interface{})
+				if !ok {
+					continue
+				}
+
+				containerInfoMap := make(map[string]string)
+				containerName := containerMap["name"].(string)
+				containerInfoMap["image"] = containerMap["image"].(string)
+				containerInfoMap["ElementName"] = containerName
+				containerInfoMap["Computer"] = nodeName
+				containerInfoMap["PodName"] = podName
+				containerInfoMap["Namespace"] = namespace
+				containerInfoMap["CreatedTime"] = createdTime
+
+				portsValue := containerMap["ports"]
+				portsValueString := ""
+				if portsValue != nil {
+					jsonStr, err := json.Marshal(portsValue)
+					if err != nil {
+						continue
+					}
+					portsValueString = string(jsonStr)
+				}
+				containerInfoMap["Ports"] = portsValueString
+
+				cmdValue := containerMap["command"]
+				cmdValueString := ""
+				if cmdValue != nil {
+					cmdValueString = fmt.Sprintf("%v", cmdValue)
+				}
+				containerInfoMap["Command"] = cmdValueString
+
+				//TODO: Remove this DEAD CODE as it was only used by replica set for Windows data
```

Comment:
TODOs can be cleaned up.

Reply:
Keeping this TODO as we might as well remove this when we are cleaning up code and everything has been moved to fluent-bit and windows DS